### PR TITLE
fix: Fix contract codes fetching for zksync chain type

### PR DESCRIPTION
**Pull request by nikitosing**
## Motivation
If internal txs indexing is enabled on ZkSync chain, then most likely deployed smart contract will inherit deployed bytecode from internal tx, but for some reasons on zksync it differs from actual deployed bytecode, similar situation with creation input. Example: [0x639ba9937e84F4dBfc6d25F972D80fE5a85d58da](https://zksync.blockscout.com/address/0x639ba9937e84F4dBfc6d25F972D80fE5a85d58da?tab=contract)
**Contract creation code**
`0x`
**Deployed ByteCode**
`0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000`

## Changelog
- Don't fetch contract code from internal transactions if chain_type == zksync
- Added backfiller, to refetch contract codes
- Added new field (`contract_code_recollected`) to addresses table, to track was migration performed for address or not.
- `MIGRATION_SANITIZE_DUPLICATED_LOG_INDEX_LOGS_CONCURRENCY`, `MIGRATION_SANITIZE_DUPLICATED_LOG_INDEX_LOGS_BATCH_SIZE` (forgotten in https://github.com/blockscout/blockscout/pull/10880)
- `MIGRATION_RECOLLECT_CONTRACT_CODES_CONCURRENCY`, `MIGRATION_RECOLLECT_CONTRACT_CODES_BATCH_SIZE`
https://github.com/blockscout/docs/pull/335

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/for-developers/information-and-settings/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

**Review by sl1depengwyn — APPROVED** [review]

File: config/runtime.exs [hunk 1]
```diff
@@ -619,6 +619,14 @@ config :explorer, Explorer.Migrator.RestoreOmittedWETHTransfers,
...
   timeout: ConfigHelper.parse_time_env_var("MIGRATION_RESTORE_OMITTED_WETH_TOKEN_TRANSFERS_TIMEOUT", "250ms")
 
+config :explorer, Explorer.Migrator.SanitizeDuplicatedLogIndexLogs,
+  concurrency: ConfigHelper.parse_integer_env_var("MIGRATION_SANITIZE_DUPLICATED_LOG_INDEX_LOGS_CONCURRENCY", 10),
+  batch_size: ConfigHelper.parse_integer_env_var("MIGRATION_SANITIZE_DUPLICATED_LOG_INDEX_LOGS_BATCH_SIZE", 500)
+
+config :explorer, Explorer.Migrator.RecollectContractCodes,
+  concurrency: ConfigHelper.parse_integer_env_var("MIGRATION_RECOLLECT_CONTRACT_CODES_CONCURRENCY", 5),
+  batch_size: ConfigHelper.parse_integer_env_var("MIGRATION_RECOLLECT_CONTRACT_CODES_BATCH_SIZE", 100)
+
```
Comment (vbaranov) [inline config/runtime.exs:629]: Let's add new env vars in common-blockscout.env

**Review by vbaranov — COMMENTED** [review]
"recollect" pattern is not used in the code base. However, "refetch" is commonly used when we want to re-fetch the data. Could we rename all entries of "recollect" to "refetch" for keeping consistency with the rest of the codebase?

File: config/runtime.exs (hunk 1 above)
Comment (nikitosing) [reply config/runtime.exs]: Processed in cd1f0a76654f0ee249becd9198da0e76c790b818

**Review by vbaranov — COMMENTED** [review]
Explorer tests on ZkSync JSON RPC variant are constantly failed https://github.com/blockscout/blockscout/actions/runs/11578889897/job/32266592180?pr=11055.

**Pull request merged**